### PR TITLE
`Trusted Entitlements`: add support for setting `EntitlementVerificationMode`

### PR DESCRIPTION
**Pull request by NachoSoto**
See http://rev.cat/trusted-entitlements for more information.

Depends on https://github.com/RevenueCat/purchases-hybrid-common/pull/451

### CODE REVIEW

File: src/purchases.ts [hunk 1]
```diff
@@ -292,12 +301,14 @@ export default class Purchases {
...
       throw new Error("appUserID needs to be a string");
     }
+
     RNPurchases.setupPurchases(
       apiKey,
       appUserID,
       observerMode,
       userDefaultsSuiteName,
       usesStoreKit2IfAvailable,
+      entitlementVerificationMode,
```
Comment (NachoSoto) [inline src/purchases.ts:311]: Need to verify that this is actually passing the string.

**Comment by tonidero** [comment]
This is ready for review

File: src/purchases.ts [hunk 2]
```diff
@@ -170,6 +186,7 @@ export default class Purchases {
    * @param {boolean} [observerMode=false] An optional boolean. Set this to TRUE if you have your own IAP implementation and want to use only RevenueCat's backend. Default is FALSE.
    * @param {boolean} [usesStoreKit2IfAvailable=false] DEPRECATED. An optional boolean. iOS-only. Defaults to FALSE. Setting this to TRUE will enable StoreKit2 on compatible devices.
    * We recommend not using this parameter, letting RevenueCat decide for you which StoreKit implementation to use.
+   * @param {ENTITLEMENT_VERIFICATION_MODE} [entitlementVerificationMode=ENTITLEMENT_VERIFICATION_MODE.DISABLED]
```
Comment (vegaro) [inline src/purchases.ts:189]: Should we add any explanation here?

**Review by vegaro — APPROVED** [review]

File: src/purchases.ts (hunk 2 above)
Comment (tonidero) [reply src/purchases.ts]: Good point! Added in https://github.com/RevenueCat/react-native-purchases/pull/678/commits/bfb8c2b19acbca3c6c9f36149b232f1cefa6e310

**Pull request merged**